### PR TITLE
fix validation

### PR DESCRIPTION
**Pull request by ling-tx**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by ling-tx** [comment]
I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer.

**Comment by v-prasadboke** [comment]
Hello @ling-tx please change the text Azure Sentinel to Microsoft Sentinel in create ui and maintemplate file

File: Solutions/IronNet IronDefense/Package/createUiDefinition.json [hunk 1]
```diff
@@ -144,12 +194,277 @@
...
+              }
+            }
+          },
+          {
+            "name": "appregistration-link2",
+            "type": "Microsoft.Common.TextBlock",
+            "options": {
+              "link": {
+                "label": "Microsoft Identity Platform Overview",
+                "uri": "https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-overview"
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Package/createUiDefinition.json:230]: please remove en-us from this link

**Comment by v-prasadboke** [comment]
And can you please describe what content you have added/modified in this solution. If yes modified/added or removed please commit those changes as well.

And please upload the zip package as well which contains the createui and maintemplate file only.

**Comment by ling-tx** [comment]
may I know what these errors mean?
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json
Could not identify JSON file as mainTemplate.json or createUiDefinition.json. Skipping. File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json
Could not identify JSON file as mainTemplate.json or createUiDefinition.json. Skipping. File path: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Workbooks/IronDefenseAlertDashboard.json
Could not identify json file as a Main Template. Skipping File path: Solutions/IronNet IronDefense/Workbooks/IronDefenseAlertDashboard.json
Could not identify JSON file as mainTemplate.json or createUiDefinition.json. Skipping. File path: Solutions/IronNet IronDefense/Workbooks/IronDefenseAlertDashboard.json

File: Solutions/IronNet IronDefense/Analytic Rules/IronDefense_Detection_Query.yaml [hunk 2]
```diff
@@ -49,7 +49,7 @@ customDetails:
   AnalystRating: LogSeverity
   AlertCreatedTime: TimeGenerated
   IronVueUrl: IronVueUrl
-  IronDefenseAlertId: DeviceCustomString4
+  IronDefenseAlertIdentifier: DeviceCustomString4
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Analytic Rules/IronDefense_Detection_Query.yaml:52]: Hello @ling-tx please decrease the number of character of this customdetail '**IronDefenseAlertIdentifier**', it should not exceed a limit of 20 characters.

**Comment by v-prasadboke** [comment]
Please Repackage the solution using V2 tool. This solution is not packaged properly i recommend you to repackage the solution using V2 tool. 
Please go through this Readme file for proper instructions. 
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md

File: Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
-  "Data Connectors": [
-    "DataConnectors/IronNetIronDefense.json"
+  "Description": "The [IronNet Collective Defense](https://www.ironnet.com/what-is-collective-defense) solution enables ingestion of IronDefense alerts, events, and IronDome notifications into Microsoft Sentinel, enabling Microsoft Sentinel to utilize IronDefense's behavioral analytics and the IronDome community to quickly identify threats in your enterprise network.\n\n**Underlying Microsoft Technologies used:**\n\nThis solution takes a dependency on the following technologies, and some of these dependencies either may be in [Preview](https://azure.microsoft.com/support/legal/preview-supplemental-terms/) state or might result in additional ingestion or operational costs:\n\n a. [Agent-based log collection (CEF over Syslog)](https://docs.microsoft.com/azure/sentinel/connect-common-event-format)",
+  "Workbooks": [
+    "Workbooks/IronDefenseAlertDashboard.json",
+    "Workbooks/IronDefenseAlertDetails.json"
+  ],
+  "WorkbookDescription": [
+     "Dashboard to view IronDefense alerts",
+     "View IronDefense alert details from the IronDefense Alerts dashboard"
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json:12]: Please this whole parameter of workbook description, it should not be included in the input file

**Comment by v-prasadboke** [comment]
Please Repackage the solution using V2 tool. This solution is not packaged properly i recommend you to repackage the solution using V2 tool.
Please go through this Readme file for proper instructions.
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md

**Comment by ling-tx** [comment]
> https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md

After seeing your comment, I notice there is v2 tools. Originally I use createSolution.ps1. 
I tried to use createSolutionV2.ps1 to create solutions but found the generated solution is much different from v1
Is it required to use v2 tool? If using v2 tool, does it mean I need to modify my json files?

**Comment by v-prasadboke** [comment]
> > https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/README.md
> 
> After seeing your comment, I notice there is v2 tools. Originally I use createSolution.ps1. I tried to use createSolutionV2.ps1 to create solutions but found the generated solution is much different from v1 Is it required to use v2 tool? If using v2 tool, does it mean I need to modify my json files?

No @ling-tx you dont need to modify your json file, just commit the changes which i asked for input file."
After that repackage the solution using v2 tool

**Comment by ling-tx** [comment]
@v-prasadboke I tried to verified new solution generated by v2 tool by "Deploy a Custom Template" on the portal, it deployed failed. 
compared v1 and v2 generated solutions, they are quite different. that's why I confused. 
v1 solution:
Screenshot 2023-06-27 at 10 19 25 AM
v2 solution:
Screenshot 2023-06-27 at 10 20 34 AM
v2 solution lack some necessary parameters. I use the same json file.

**Comment by v-prasadboke** [comment]
Hello @ling-tx let me look into this

File: Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json (hunk 3 above)
Comment (v-prasadboke) [reply Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json:12]: Hello @ling-tx please work on this change and repackage the solution again

**Comment by v-prasadboke** [comment]
https://github.com/Azure/Azure-Sentinel/pull/8328/files/56a05b42156c20b10ed7fdc711a3669e347f7364#r1241890466

File: Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json (hunk 3 above)
Comment (ling-tx) [reply Solutions/IronNet IronDefense/Data/Solution_IronNetIronDefense.json]: I remove the workbook description and repackage again

**Comment by v-prasadboke** [comment]
Hello @ling-tx as per new guidelines issued by Microsoft Sentinel, please repackage the solution using V3 tool 
Please go through this readme file on how to use the V3 tool

**Comment by v-prasadboke** [comment]
Hello @ling-tx any updates on the above

**Comment by v-prasadboke** [comment]
Hello @ling-tx how were you able to repackage the solution with v3 tool, because your branch is not updated with master and i cant see the V3 folder in your branch.

**Comment by v-prasadboke** [comment]
Please repackage the solution With v3 and commit the changes, i wont be able to get what issues you are facing with the deployement

**Comment by ling-tx** [comment]
Hello, I upload the solution with v3. if deploy mainTemplate.json, it doesn't create 2 log app as expected

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json [hunk 4]
```diff
@@ -2,67 +2,28 @@
...
-    "comments": "Enrich IronNet Alerts based on the corresponding Sentinel Incident updates via IronNet Endpoints",
+    "comments": "Enrich IronNet Alerts based on the corresponding Microsoft Sentinel Incident updates via IronNet Endpoints",
     "author": "IronNet"
   },
   "parameters": {
-    "IronApiUsername": {
-      "type": "String",
+    "PlaybookName": {
+      "type": "string",
+      "repeat": "false",
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json:11]: hello @ling-tx any use of the "repeat": "false"

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json [hunk 5]
```diff
@@ -2,67 +2,28 @@
...
   "variables": {
-    "azuresentinelConnectionName": "[concat('azuresentinel-', parameters('Playbook Name'))]"
+    "azuresentinelConnectionName": "[concat('azuresentinel-', parameters('PlaybookName'))]"
   },
   "resources": [
     {
       "type": "Microsoft.Web/connections",
       "apiVersion": "2016-06-01",
-      "name": "[variables('azuresentinelConnectionName')]",
+      "name": "[variables('playbook2-azuresentinelConnectionName')]",
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json:26]: Please revert 'playbook2-azuresentinelConnectionName' to 'azuresentinelConnectionName'

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json [hunk 6]
```diff
@@ -90,42 +51,50 @@
...
+              "type": "String"
             },
+            "password": {
+							"defaultValue": "[parameters('playbook1-IronApiPassword')]",
+							"type": "SecureString"
+						},
             "client_id": {
-              "defaultValue": "[parameters('ClientId')]",
-              "type": "SecureString"
+              "defaultValue": "[parameters('playbook1-AzureClientId')]",
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json:71]: Prameters are missing from the Parameters section, please add them in playbook paramters

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json [hunk 7]
```diff
@@ -2,67 +2,28 @@
...
-      }
-    },
-    "Playbook Name": {
-      "type": "String",
-      "metadata": {
-        "description": "Playbook name to test the IronNet API"
-      }
+				"label": "Playbook Name",
+				"description": "Playbook name to test the IronNet API"
+			}
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json:16]: Please add IronDefenseUrl, IronApiUsername, IronApiPassword, AzureClientId, AzureClientSecret, workspace in paramters section

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json [hunk 8]
```diff
@@ -98,73 +105,81 @@
...
-							"type": "SecureString"
+						"tenant_id": {
+							"defaultValue": "[subscription().tenantId]",
+							"type": "String"
 						},
-						"Password": {
-							"defaultValue": "[parameters('IronApiPassword')]",
-							"type": "SecureString"
+						"irondefense_url": {
+							"defaultValue": "[parameters('playbook1-IronDefenseUrl')]",
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json:113]: Please remove playbook1 from all the "defaultvalue" section
Paramters used in "defaultvalue" should be matching with the parameters

**Comment by v-prasadboke** [comment]
Please add workbookmetadata of workbooks in " Azure-Sentinel\Tools\Create-Azure-Sentinel-Solution\V2\WorkbookMetadata\WorkbooksMetadata.json "

**Comment by v-prasadboke** [comment]
After all the changes done, please repackage the solution again

**Comment by v-prasadboke** [comment]
Hello @ling-tx any updates On the above changes requested

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json (hunk 6 above)
Comment (ling-tx) [reply Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateSentinelIncidents/azuredeploy.json:71]: They are the same as playbook1. Is there possible that I can share the same parameter for playbook1 and playbbok2?

**Comment by ling-tx** [comment]
I fix the comments and create new package. 
The UI of playbook still lack necessary input parameter 
Screenshot 2023-07-20 at 4 29 27 PM
When verify mainTemplate.json, I see warning
Screenshot 2023-07-20 at 4 33 54 PM

**Comment by v-prasadboke** [comment]
Hello @ling-tx, 
Thanks for the updates.
Will be investigating the issue you are facing, You can expect an update till 25 Jul 2023.

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json [hunk 9]
```diff
@@ -2,65 +2,69 @@
...
 	"metadata": {
-		"comments": "Updates IronDefense alert workflow status and rating via IronAPI when the corresponding Sentinel Incident is updated.",
+		"comments": "Updates IronDefense alert workflow status and rating via IronAPI when the corresponding Microsoft Sentinel Incident is updated.",
 		"author": "IronNet"
 	},
 	"parameters": {
 		"IronApiUsername": {
-			"type": "String",
+			"type": "string",
+			"repeat": "false",
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json:11]: Hello @ling-tx please can you tell if there is any need of this parameter "repeat":

**Comment by v-prasadboke** [comment]
Hello @ling-tx, While custom deploying the playbook I got this error.
image

Can you please have a look at this.

**Comment by v-prasadboke** [comment]
And also if possible can we schedule a call regarding this. I work in IST time zone, can you please suggest a time as per your availability. I'm usually available between 2:00 PM to 6:00 PM ( IST time zone ).

Thank you,
Prasad Boke.

**Comment by ling-tx** [comment]
Hello, I am in CST timezone. Is it ok we meet at 5:30PM IST timezone?

**Comment by v-prasadboke** [comment]
Thank you @ling-tx, I'll schedule a call for 5:30 PM IST which is 7:00 AM CST and send you the invitation link for the meet.

**Comment by ling-tx** [comment]
Hello, my email is yilin.chen@ironnetcybersecurity.com. thanks

**Comment by v-prasadboke** [comment]
Thank you @ling-tx, Will be scheduling a call for 5:30 PM IST 27 Jul 2023

**Comment by v-prasadboke** [comment]
Hello @ling-tx, as discussed in the call. Please go with the changes, Im trying to find the playbook which I was gonna share it with you. 
I'll share it ASAP as i find it.
Thank you.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, we are still examining the issue. You can expect an update till 03 Aug 2023.

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json [hunk 10]
```diff
@@ -98,73 +102,81 @@
...
 						},
-						"ShareCommentWithIronDome": {
-							"defaultValue": false,
+						"share_comment_with_irondome": {
+							"defaultValue": "[parameters('ShareCommentWithIrondome')]",
 							"type": "Bool"
+						},
+						"microsoft_login_uri": {
+							"type": "string",
+							"defaultValue": "[concat('login.microsoft','online.com')]"
```
Comment (v-prasadboke) [inline Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json:159]: Hello @ling-tx, Can you please declare a paramter for this and use it here, In that parameter's default value set as [concat('login.microsoft','online.com')]

Same is the case with other concanited strings as well.  Thanks

**Comment by v-prasadboke** [comment]
And please also commit the changes which were requested earlier. Thank you.

File: Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json (hunk 10 above)
Comment (ling-tx) [reply Solutions/IronNet IronDefense/Playbooks/IronNet_UpdateIronDefenseAlerts/azuredeploy.json]: I declare parameters but still have concat error

**Comment by v-prasadboke** [comment]
Hello @ling-tx, I've scheduled a call for tomorrow. Scheduled the meet for 7:00 AM (CST) 07 Aug 2023 Tuesday.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Thank you for your time. And as I ensured I'll look at this issue and update you about the same soonest.

**Comment by thao2** [comment]
@v-prasadboke,

@ling-tx is on PTO until Monday Aug 21.  I can cover for her if you need to schedule a meeting before then.

**Comment by v-prasadboke** [comment]
Hello there, @thao2. Thank you for informing me. I'll contact her once when she returns back.

**Comment by ling-tx** [comment]
hello, @v-prasadboke, I am available tomorrow. thanks.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Sorry for the inconvenience. But due to some other conflicts it was not possible for me to attend the call. 
Is it okay if we reschedule this call for tomorrow.

**Comment by ling-tx** [comment]
@v-prasadboke tomorrow is fine. please book the meeting. thanks.

**Comment by v-prasadboke** [comment]
Thank you @ling-tx, Scheduling the meet for tomorrow.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, As discussed in the call. Try generating the playbook arm template and continue with commits.
Please reach out to us if anything needed.
Thank you very much.

**Comment by ling-tx** [comment]
The Playbook_ARM_Template_Generator.ps1 can't work on mac platform
Screenshot 2023-08-23 at 2 22 54 PM

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Thank you for bringing this upfront. As we have scheduled a call we will have this discussed in it.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, As discussed in the call regarding the issue you are facing we will have someone from the repackaging tool team to look in this.
Thanks, 
Prasad.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, We are still discussing the issue you are facing. Will update you till tomorrow about this.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, The Playbook parameters wont get displayed on the custom deploy page while deploying the maintemplate.
As for the parameters, It will be visible after deploying the maintemplate and selecting the playbook. You can add parameters for the playbook, on create playbook page.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Reason might be Playbook metadata which needs to filled. If all the entities of playbook metadata are empty then the automation blade of azure portal fails to load. 
I would recommend to fill all the playbook metadata entities and then repackage the solution.
Try Deploying the maintemplate after this.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Hope you are doing well. As communicated on the chat, Please continue filling all the metadata details of the playbook and thus with repackaging the solution after filling the details. 
Thank you.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, It's been a while I haven't heard from you. Hope you are fine. 
Also please provide some updates to the above comment.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Please respond to the above comments.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Please respond to the above comments.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Please respond to the above comments.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Hope you are good. Just wanted to know if there's any updates on the above discussions.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, It's been a long time we haven't heard anything from you. Hope you are doing well, Just wanted to check if there's any progress on the above requested changes.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ling-tx, Please respond to the above mentioned comments.
Thanks.

**Comment by v-prasadboke** [comment]
We wanted to check on the status of PR #8328 . PR is pending for more than 30 days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 7 business days, we will close this PR. Thank you for your cooperation.

**Comment by thao2** [comment]
Hello @v-prasadboke,

First, I'd like to say your dedication with the frequent postings you've made with no response by Ling to this issue is extremely commendable.  Second, I wanted to let you know that IronNet furloughed its employees on Sept 2, and ceased operations yesterday, Sept 29th, 2023.  More info here:  https://ir.ironnet.com/financials/all-sec-filings.

I just wanted to make a courtesy post so you don't waste any more time on this, but I don't expect this issue to be picked up again (I don't speak for IronNet).

**Comment by thao2** [comment]
Hello @v-prasadboke,

First, I'd like to say your dedication with the frequent postings you've made with no response by Ling to this issue is extremely commendable. Second, I wanted to let you know that IronNet furloughed its employees on Sept 2nd, 2023, and ceased operations yesterday, Sept 29th, 2023. I don't speak for IronNet, but wanted to let you know so you don't waste any more time on this as I don't expect this issue to be picked up again

More info here: https://ir.ironnet.com/financials/all-sec-filings.

**Comment by v-prasadboke** [comment]
Hello @thao2, thank you for update, we are sorry to hear this about IronNet.inc. We are closing this PR as per our standard procedure. Thank you for you co-operation.

**Pull request closed without merging**